Allow elasticsearch to update member hosts in unicast mode from bind peers

elliott-davis opened the pull request:
Signed-off-by: Elliott Davis <edavis@chef.io>

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

elliott-davis commented:
```Elliotts-MacBook-Pro:~ elliott$ docker run -it --rm edavis/elasticsearch --peer 172.17.0.3 --bind elasticsearch:elasticsearch.default
hab-sup(MN): Starting edavis/elasticsearch
hab-sup(TP): Child process will run as user=hab, group=hab
hab-sup(GS): Supervisor 172.17.0.4: 2736cf4a-c050-4a14-83ce-9940a23186dd
hab-sup(GS): Census elasticsearch.default: 17ec324e-f3ca-48ca-bd78-74d3f528802a
hab-sup(GS): Starting inbound gossip listener
hab-sup(GS): Joining gossip peer at 172.17.0.3:9634
hab-sup(GS): Starting outbound gossip distributor
hab-sup(GS): Starting gossip failure detector
hab-sup(CN): Starting census health adjuster
hab-sup(SC): Updated elasticsearch.yml
hab-sup(SC): Updated logging.yml
hab-sup(SC): Updated health_check
hab-sup(TP): Restarting because the service config was updated via the census
elasticsearch(SV): Starting
elasticsearch(O): [2016-11-27 23:06:45,631][WARN ][bootstrap                ] unable to install syscall filter: seccomp unavailable: your kernel is buggy and you should upgrade
elasticsearch(O): [2016-11-27 23:06:45,637][WARN ][bootstrap                ] Unable to lock JVM Memory: error=12,reason=Cannot allocate memory
elasticsearch(O): [2016-11-27 23:06:45,637][WARN ][bootstrap                ] This can result in part of the JVM being swapped out.
elasticsearch(O): [2016-11-27 23:06:45,637][WARN ][bootstrap                ] Increase RLIMIT_MEMLOCK, soft limit: 83968000, hard limit: 83968000
elasticsearch(O): [2016-11-27 23:06:45,637][WARN ][bootstrap                ] These can be adjusted by modifying /etc/security/limits.conf, for example:
elasticsearch(O): 	# allow user 'hab' mlockall
elasticsearch(O): 	hab soft memlock unlimited
elasticsearch(O): 	hab hard memlock unlimited
elasticsearch(O): [2016-11-27 23:06:45,637][WARN ][bootstrap                ] If you are logged in interactively, you will have to re-login for the new limits to take effect.
elasticsearch(O): [2016-11-27 23:06:45,800][INFO ][node                     ] [Leonus] version[2.4.1], pid[78], build[c67dc32/2016-09-27T18:57:55Z]
elasticsearch(O): [2016-11-27 23:06:45,800][INFO ][node                     ] [Leonus] initializing ...
elasticsearch(O): [2016-11-27 23:06:46,342][INFO ][plugins                  ] [Leonus] modules [reindex, lang-expression, lang-groovy], plugins [], sites []
elasticsearch(O): [2016-11-27 23:06:46,368][INFO ][env                      ] [Leonus] using [1] data paths, mounts [[/hab/svc/elasticsearch/data (/dev/vda2)]], net usable_space [35.1gb], net total_space [59gb], spins? [possibly], types [ext4]
elasticsearch(O): [2016-11-27 23:06:46,368][INFO ][env                      ] [Leonus] heap size [989.8mb], compressed ordinary object pointers [true]
elasticsearch(O): [2016-11-27 23:06:48,030][INFO ][node                     ] [Leonus] initialized
elasticsearch(O): [2016-11-27 23:06:48,030][INFO ][node                     ] [Leonus] starting ...
elasticsearch(O): [2016-11-27 23:06:48,142][INFO ][transport                ] [Leonus] publish_address {172.17.0.4:9300}, bound_addresses {172.17.0.4:9300}
elasticsearch(O): [2016-11-27 23:06:48,151][INFO ][discovery                ] [Leonus] elasticsearch/4GzFfrzqQEq8o0IbgCI2cw
elasticsearch(O): [2016-11-27 23:06:51,250][INFO ][cluster.service          ] [Leonus] detected_master {Doctor Demonicus}{meQGHCctTbWvgMH6moOWlA}{172.17.0.3}{172.17.0.3:9300}{max_local_storage_nodes=1, master=true}, added {{Doctor Demonicus}{meQGHCctTbWvgMH6moOWlA}{172.17.0.3}{172.17.0.3:9300}{max_local_storage_nodes=1, master=true},}, reason: zen-disco-receive(from master [{Doctor Demonicus}{meQGHCctTbWvgMH6moOWlA}{172.17.0.3}{172.17.0.3:9300}{max_local_storage_nodes=1, master=true}])
elasticsearch(O): [2016-11-27 23:06:51,310][INFO ][http                     ] [Leonus] publish_address {172.17.0.4:9200}, bound_addresses {172.17.0.4:9200}
elasticsearch(O): [2016-11-27 23:06:51,310][INFO ][node                     ] [Leonus] started
elasticsearch(O): [2016-11-27 23:07:36,129][INFO ][discovery.zen            ] [Leonus] master_left [{Doctor Demonicus}{meQGHCctTbWvgMH6moOWlA}{172.17.0.3}{172.17.0.3:9300}{max_local_storage_nodes=1, master=true}], reason [shut_down]
elasticsearch(O): [2016-11-27 23:07:36,130][WARN ][discovery.zen            ] [Leonus] master left (reason = shut_down), current nodes: {{Leonus}{4GzFfrzqQEq8o0IbgCI2cw}{172.17.0.4}{172.17.0.4:9300}{max_local_storage_nodes=1, master=true},}
elasticsearch(O): [2016-11-27 23:07:36,130][INFO ][cluster.service          ] [Leonus] removed {{Doctor Demonicus}{meQGHCctTbWvgMH6moOWlA}{172.17.0.3}{172.17.0.3:9300}{max_local_storage_nodes=1, master=true},}, reason: zen-disco-master_failed ({Doctor Demonicus}{meQGHCctTbWvgMH6moOWlA}{172.17.0.3}{172.17.0.3:9300}{max_local_storage_nodes=1, master=true})
^Celasticsearch(O): [2016-11-27 23:07:39,400][INFO ][node                     ] [Leonus] stopping ...```

eeyun commented:
[core/elasticsearch/2.4.1/20161128183355](https://app.habitat.sh/#/pkgs/core/elasticsearch/2.4.1/20161128183355) has been built and uploaded to the depot. :sparkling_heart:

![](https://media.giphy.com/media/l3vRhFcpkE6CgYKuA/giphy.gif)
